### DISCUSSION TITLE
Fail2Ban - Registration not allowed or user already exists

### DISCUSSION THREAD

**Discussion by kramttocs** [dani-garcia/vaultwarden#5234] [Ideas]
Setup Fail2Ban and it's working great for invalid login attempts. I'd also like to block registration attempts since it's disabled.
I believe the only thing needed would be adding the IP to:

[2024-11-25 21:15:43.885][vaultwarden::api::core::accounts][ERROR] Registration not allowed or user already exists

That way a filter can be created for it.

**Comment by stefan0xC** [comment]
> I believe the only thing needed would be adding the IP to:

Are you using a reverse proxy? That way you could filter the access log for lines containing `POST /identity/accounts/register`.

**Reply by kramttocs** [reply]
Maybe I don't have a good understanding of fail2ban - does it not need the IP in the log entry?

Edit: now I follow. Do it at the reverse proxy level. Yes that would be possible but having the IP in the VW log stills seems ideal

**Reply by BlackDex** [reply]
Which the reverse proxy should provide for every request
